packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -267,7 +269,7 @@ const renderApp = (args) => {
     const windowGlobals = {
         __DEVICE_TYPE__: deviceType,
         __PRELOADED_STATE__: appState,
-        __ERROR__: error,
+        __ERROR__: isProduction ? undefined : error,
```

Comment:
This currently sets error to undefined if the deploy target is 'production'. Using undefined here produces a 'cannot read properties of undefined' error somewhere so would using an empty string be better?

Reply:
Ben asked me for an opinion! Some pointers:

1. The `process.env.DEPLOY_TARGET` var is unfortunately just a name. You can't use it reliably to find out what a production environment is on MrT, but a solution is in the works ([massive internal thread here], if you're curious).
2. Given (1), and that this is a security issue, I would probably base this behaviour on [`isRemote`]. That would mean you can't see stack traces on any deployed instance.
3. If, in future, you really want to show these stack traces on non-production deployed instances, reach out to the MrT team.
4. Don't ditch the error object entirely – we just want to get rid of the stack trace. But I think it comes with a status and a message that you would still want to show!

[massive internal thread here]: https://mobify.slack.com/archives/C8YDDMKFZ/p1637878095039600
[`isRemote`]: https://github.com/SalesforceCommerceCloud/pwa-kit/blob/59b6a392f51ec7f7336e12434238169f1b27eebb/packages/pwa-kit-react-sdk/src/utils/ssr-server.js#L37

Reply:
@vcua-mobify Seems like there's a bit of confusion about whether you even wanted to use "production" targets here at all! https://mobify.slack.com/archives/C8YDDMKFZ/p1637956473109800

Reply:
Good catch @olibrook about not throwing the baby out with the bath water. We could probably do something like:

```
if (error && isRemote()) {
    delete error.stack
}
```

Reply:
@vcua-mobify ☝️  has not been resolved (point number 4)

Reply:
Point #4 has been addressed. Only the stack is removed in remote environments.